File: Server/Models/Alert.cs
namespace Server.Models;

public class Alert
{
    public string Id { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Severity { get; set; } = "Low"; // Critical, High, Medium, Low
    public string Category { get; set; } = "General";
    public string Status { get; set; } = "New"; // New, In Progress, Resolved, Escalated, Dismissed
    public DateTime Timestamp { get; set; }
    public string SourceIp { get; set; } = string.Empty;
    public string DestinationHost { get; set; } = string.Empty;
    public string UserAccount { get; set; } = string.Empty;
    public string MitreTechnique { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public List<string> RawLogs { get; set; } = new();

    // AI Enrichment fields
    public string AiSummary { get; set; } = string.Empty;
    public int AiConfidenceScore { get; set; }
    public string AiRiskAnalysis { get; set; } = string.Empty;
    public List<string> AiRecommendedActions { get; set; } = new();
}

File: Client/src/types/alert.ts
export interface Alert {
  id: string;
  title: string;
  severity: 'Critical' | 'High' | 'Medium' | 'Low';
  category: string;
  status: 'New' | 'In Progress' | 'Resolved' | 'Escalated';
  timestamp: string;
  sourceIp: string;
  destinationHost: string;
  userAccount: string;
  mitreTechnique: string;
  description: string;
  rawLogs: string[];
  
  // Pola ewaluacji (testu)
  isThreat?: boolean;
  correctAction?: string;

  // Pola wzbogacone przez AI (dla widoku Z AI)
  aiSummary?: string;
  aiConfidenceScore?: number;
  aiRiskAnalysis?: string;
  aiRecommendedActions?: string[];
}

export interface OperatorDecision {
  alertId: string;
  actionTaken: string;
  decisionTimeSeconds: number;
  timestamp: string;
  isThreat?: boolean;
  correctAction?: string;
  category?: string;
}

export interface TestSession {
  sessionId?: string;
  operatorName: string;
  mode: 'NoAI' | 'WithAI';
  startTime: string;
  alertsHandledCount: number;
  totalDurationSeconds: number;
  decisions: OperatorDecision[];
}

export interface UserSession {
  token: string;
  username: string;
  email: string;
  role: string;
  databaseMode: string;
}
